citesphere/src/main/java/edu/asu/diging/citesphere/web/user/authorities/helper/AuthorityEntryControllerHelper.java

```diff
@@ -0,0 +1,61 @@
+package edu.asu.diging.citesphere.web.user.authorities.helper;
+
+import java.util.List;
+import java.util.ListIterator;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.citesphere.model.authority.IAuthorityEntry;
+import edu.asu.diging.citesphere.web.user.FoundAuthorities;
+
+@Service
+public class AuthorityEntryControllerHelper {
+	
+    @Value("${_conceptpower_authority_search_keyword}")
+    private String conceptpowerSearckKeyword;
+    
+    public String[] getAuthoritySearchStrings(String firstName, String lastName) {
+	  	
+		String conceptpowerSearchString = "";
+		String databaseSearchString = "";
+		
+    	if(firstName!=null && lastName!=null) {
+    		conceptpowerSearchString = conceptpowerSearckKeyword+"%25"+firstName+"%25"+lastName;
+    		databaseSearchString = "%"+lastName+"%"+firstName+"%";
+    	}
+    	else if(firstName==null) {
+    		conceptpowerSearchString = conceptpowerSearckKeyword+"%25"+lastName;
+    		databaseSearchString = "%"+lastName+"%";
+    	}
+    	else {
+    		conceptpowerSearchString =  conceptpowerSearckKeyword+"%25"+firstName;
+    		databaseSearchString = "%"+firstName+"%";
+    	}
+    	
+    	return new String[] {conceptpowerSearchString, databaseSearchString};
```

Comment:
for urlencoding there are libraries that do that. for the database, can't you use spring data's containing keyword to automatically generate the query? https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.query-creation